sklearn/feature_selection/sequential_feature_selector.py

```diff
@@ -0,0 +1,332 @@
+                                 ' elements a min and a max value.')
+
+            if self.n_features_to_select[0] not in range(1, X.shape[1] + 1):
+                raise ValueError('n_features_to_select tuple min value must'
+                                 ' be in range(1, X.shape[1]+1).')
+
+            if self.n_features_to_select[1] not in range(1, X.shape[1] + 1):
+                raise ValueError('n_features_to_select tuple max value must'
+                                 ' be in range(1, X.shape[1]+1).')
+
+            if self.n_features_to_select[0] > self.n_features_to_select[1]:
+                raise ValueError('The min n_features_to_select value must be'
+                                 ' smaller than the max'
+                                 ' n_features_to_select value.')
+
+        if isinstance(self.n_features_to_select, tuple):
+            select_in_range = True
+        else:
+            select_in_range = False
+            k_to_select = self.n_features_to_select
+
+        self._scorer = check_scoring(self.estimator, scoring=self.scoring)
+
+        cloned_estimator = clone(self.estimator)
+
+        self._n_features = X.shape[1]
+        self.subsets_ = {}
+        orig_set = set(range(self._n_features))
+        if self.forward:
+            if select_in_range:
+                k_to_select = self.n_features_to_select[1]
+            k_idx = ()
+            k = 0
+        else:
+            if select_in_range:
+                k_to_select = self.n_features_to_select[0]
+            k_idx = tuple(range(self._n_features))
+            k = len(k_idx)
+            k_score = self._calc_score(X, y, k_idx, cloned_estimator)
+            self.subsets_[k] = {
+                'feature_subset_idx': k_idx,
+                'cv_scores': k_score,
+                'avg_score': np.mean(k_score)
+                }
+
+        best_subset = None
+        k_score = 0
+
+        while k != k_to_select:
+            prev_subset = set(k_idx)
+            if self.forward:
+                k_idx, k_score, cv_scores = self._inclusion(
+                    orig_set=orig_set,
+                    subset=prev_subset,
+                    X=X,
+                    y=y,
+                    estimator=cloned_estimator
+                )
+            else:
+                k_idx, k_score, cv_scores = self._exclusion(
+                    feature_set=prev_subset,
+                    X=X,
+                    y=y,
+                    estimator=cloned_estimator
+                )
+
+            k = len(k_idx)
+            if k not in self.subsets_ or (self.subsets_[k]['avg_score'] <
+                                          k_score):
+                self.subsets_[k] = {
+                    'feature_subset_idx': k_idx,
+                    'cv_scores': cv_scores,
+                    'avg_score': k_score
+                }
+
+        if select_in_range:
+            max_score = float('-inf')
+            for k in self.subsets_:
```

Comment:
Why can't we use range here?

Reply:
instead of `for k in self.subsets_:`? what would be the advantage?